### PR TITLE
Feature: grateful skeleton (KIDS-1406)

### PR DESCRIPTION
**Pull request by TammiLion**
https://linear.app/givt/issue/KIDS-1418/create-an-overarching-logic-thats-connects-the-components-aka-a

## Release Notes

- **New Features**
	- Introduced the `GratefulCubit` for managing gratitude-related state and interactions.
	- Added `GratefulRecommendationsRepository` for handling gratitude recommendations.
	- Implemented `GratefulScreen` for displaying gratitude content with avatar interactions.
	- Created `GratefulAvatarBar` widget for user engagement through avatar selection.
	- Added functionality to save gratitude interests for superhero profiles.

- **Enhancements**
	- Expanded `GameProfile` to include `gratitudeInterests`.
	- Improved dependency injection for better organization and readability.

- **Documentation**
	- Updated class and method definitions to reflect new functionalities and structures.

### CODE REVIEW

File: lib/features/family/features/reflect/domain/grateful_recommendations_repository_impl.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class GratefulRecommendationsRepositoryImpl
+    implements GratefulRecommendationsRepository {
+
+  GratefulRecommendationsRepositoryImpl(this._organisationsRepository);
+
+  final OrganisationsRepository _organisationsRepository;
+  final Map<GameProfile, List<Organisation>> _gratefulRecommendations = {};
+
+  @override
+  Future<Map<GameProfile, List<Organisation>>>
```
Comment (MaikelStuivenberg) [inline lib/features/family/features/reflect/domain/grateful_recommendations_repository_impl.dart:15]: This method is going to fetch and store the data in the repository I think? :) 
Should it have a return type or just void?

File: lib/features/family/features/reflect/domain/reflect_and_share_repository.dart [hunk 2]
```diff
@@ -19,6 +20,11 @@ class ReflectAndShareRepository {
 
   final List<String> _usedSecretWords = [];
 
+  void saveGratitudeInterestsForCurrentSuperhero(List<Tag>? interests) {
+    _selectedProfiles[_getCurrentSuperHeroIndex()] =
```
Comment (MaikelStuivenberg) [inline lib/features/family/features/reflect/domain/reflect_and_share_repository.dart:24]: Somewhere also a translation need to happen between the 6 categories and the tag we know from the Charity Finder. Is that going to be this place? Or the cubit thats gonna call this method?

File: lib/features/family/features/reflect/domain/grateful_recommendations_repository_impl.dart (hunk 1 above)
Comment (TammiLion) [reply lib/features/family/features/reflect/domain/grateful_recommendations_repository_impl.dart:15]: Yes. Could also be void but then I would rename it.

File: lib/features/family/features/reflect/domain/reflect_and_share_repository.dart (hunk 2 above)
Comment (TammiLion) [reply lib/features/family/features/reflect/domain/reflect_and_share_repository.dart:24]: Not this place, the cubit that calls it. Wether its 6/12 or whatever amount of categories, at the end of the day I think it still just a list of interests, which is all we need for the domain.

**Review by MaikelStuivenberg — APPROVED** [review]

File: lib/features/family/features/reflect/domain/reflect_and_share_repository.dart (hunk 2 above)
Comment (Daniela510) [reply lib/features/family/features/reflect/domain/reflect_and_share_repository.dart:24]: I have the catgories containing the tags hardcoded, maybe we take care of this later

**Review by Daniela510 — APPROVED** [review]

**Pull request merged**